src/mup_code

```diff
@@ -35,8 +35,14 @@ DIRS="\
  utilities\
 "
 
-for i in $DIRS
-do
-    echo "Writing multiprecision code for $i ..."
-    (cd $i; ./mup_code)
-done
+# Function to process a single directory
+process_directory() {
+    local dir="$1"
+    echo "Writing multiprecision code for $dir ..."
+    (cd "$dir" && ./mup_code)
+}
+
+export -f process_directory
+
+# Use xargs for parallel execution with number of CPU cores
+echo "$DIRS" | tr ' ' '\n' | xargs -P "$(nproc)" -I {} bash -c 'process_directory "$1"' _ {}
```

Comment:
Is the underscore an end-of-line marker?

Reply:
This is actually a dummy var for `$0` when calling `bash -c`. It’s required so that `{}` becomes `$1` in the script and the folders get parsed correctly by `process_directory`

Reply:
After a little research on nproc, maybe it would be better to just hardcode a max proc number, say 4 or 6.  I'm not sure we get much benefit with higher numbers here anyway.

Reply:
Yes, it seems MacOS doesn't get it. Let me update

Reply:
I updated to `"${NPROC:-4}"` so it uses 4 cores by default and allow the user to overwrite it if needed